app/models/user.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class User < ApplicationRecord
+  include Authenticable
+  PASSWORD_PATTERN = /(?=.*[A-Z])(?=.*[a-z])(?=.*\d).{8,}/
+
+  before_validation :password_complexity, on: :create
+
+  def password_complexity
```

Comment:
This method must be private, then, no one can access it out of this validation ✨ 

```suggestion
  private
  
  def password_complexity
```